### DISCUSSION TITLE
umi3.x项目想要支持keep alive，有啥解决方案吗

### DISCUSSION THREAD

**Discussion by jiajiaxing1** [umijs/umi#11439] [Q&A] [answered]
查找了相关资料，可以使用 umi-plugin-keep-alive插件，但是需要在layout包一下content的内容，项目中用的是@umijs/plugin-layout，没有找到可以重写content的入口
有几个问题
1.Keep Alive是否有其他解法？还是必须使用 umi-plugin-keep-alive这个插件
2.umi3.x支持 umi-plugin-keep-alive这个插件吗
3.如果支持的话，需要重写layout，@umijs/plugin-layout是否有配置可支持重写内容区域

**Comment by fz6m** [comment]
这个问题要问一下 @xiaohuoni  ，umi 3 里 layout 插件能结合 keep alive 吗？🌹

**Reply by xiaohuoni** [reply]
一样有对应插件，可以选择用
@alitajs/keep-alive
umi-plugin-keep-alive@0.0.1-beta.30
需要注意的是 umi-plugin-keep-alive 可能需要锁定版本

**Reply by jiajiaxing1** [reply]
有无使用文档呀？ 主要是在配置layout的时候卡住了，项目中用的@umijs/plugin-layout这个插件，没有layout/index.tsx这个文件，在配置的时候不知道怎么去用组件包裹layoutd的内容区
![Uploading 企业微信截图_d922b66b-3344-4688-b451-c5db2875d92f.png…]()

**Reply by xiaohuoni** [reply]
@alitajs/keep-alive https://v2.alitajs.com/plugins/keepalive 
umi-plugin-keep-alive对应仓库的 readme 就是文档
@umijs/plugin-layout 文档 https://v3.umijs.org/zh-CN/plugins/plugin-layout#%E8%BF%90%E8%A1%8C%E6%97%B6%E9%85%8D%E7%BD%AE

新建一个 `layouts/index.tsx`

**Reply by jiajiaxing1** [reply]
新建一个layouts/index.tsx的话 是需要把原本用@umijs/plugin-layout支持的内容 全部要自己实现一遍吗
就是说用新建的layouts/index.tsx，怎么和插件@umijs/plugin-layout做关联呀

**Reply by xiaohuoni** [reply, marked as answer]
umi@3 配置里面配置 childrenRender
